### PR TITLE
[WIP] GKE cluster provisioning

### PR DESCRIPTION
**Pull request by rhs**
Add @google-cloud/container as a dependency and add a GKE module with primitives needed for
allocating, deleting, and expiring GKE clusters.

### CODE REVIEW

**Review by aosoriodw — COMMENTED** [review]
Overall really good!

Mostly questions, and one possible issue.

File: provision-cluster/gke.js [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+  // clusters that were provisioned by hand or any means other than this action. That is why we
+  // check for both the `provisioned-category` and `provisioned-lifespan` labels and ignore the
+  // cluster if they are not present and set to the correct value. We are expecting that only
+  // clusters provisioned by this action will have those labels.
+  async maybeExpireCluster(cluster, lifespanOverride) {
+    let labels = cluster.resourceLabels
+
+    let category = labels[CATEGORY_PROPERTY]
+    if (category !== EPHEMERAL) {
+      console.log(`Ignoring cluster ${cluster.name} because it has not ephemeral.`)
```
[the remark this answers is not in the document]
Comment (rhs) [reply provision-cluster/gke.js:156]: @aosoriodw You're probably right. The console.log comes from the example code I started with. That and I forgot core had special logging. ;-)

File: provision-cluster/gke.js [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+    }
+
+    if (lifespanMillis <= 0) {
+      console.log(`Keeping ${cluster.name} because the lifespan is <= 0.`)
+      return
+    }
+
+    let ageMillis = Date.now() - Date.parse(cluster.createTime)
+    if (ageMillis < lifespanMillis) {
+      console.log(`Keeping ${cluster.name} because ${ageMillis/1000}s < ${lifespanMillis/1000}s.`)
```
[the remark this answers is not in the document]
Comment (rhs) [reply provision-cluster/gke.js:184]: yes

**Review by aosoriodw — APPROVED** [review]
LGTM

**Pull request merged**